python/ray/tune/schedulers/pbt.py

```diff
@@ -397,13 +398,17 @@ def on_trial_result(self, trial_runner: "trial_runner.TrialRunner",
         if not self._synch:
             state.last_perturbation_time = time
             lower_quantile, upper_quantile = self._quantiles()
-            self._perturb_trial(trial, trial_runner, upper_quantile,
-                                lower_quantile)
-            for trial in trial_runner.get_trials():
-                if trial.status in [Trial.PENDING, Trial.PAUSED]:
-                    return TrialScheduler.PAUSE  # yield time to other trials
-
-            return TrialScheduler.CONTINUE
+            decision = TrialScheduler.CONTINUE
+            for _ in trial_runner.get_trials():
+                if _.status in [Trial.PENDING, Trial.PAUSED]:
+                    decision = TrialScheduler.PAUSE
+                    break
```

Comment:
Nit
```suggestion
            for trial in trial_runner.get_trials():
                if trial.status in [Trial.PENDING, Trial.PAUSED]:
                    decision = TrialScheduler.PAUSE
                    break
```

Reply:
Also, let's add comments for the logic here:

1) If any other trials are pending or scheduled, pause this trial to yield time to other trials
2) Overwrite decision depending on exploit outcome

Reply:
> Nit

Changed to `other_trial` to avoid shadowing `trial` in outer scope.